### PR TITLE
[C++ SDK] SDK as CMake sub project.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: client/client-multi/client-cpp/CMakeLists.txt [hunk 1]
```diff
@@ -132,7 +132,9 @@
 #
 #       Default: `0`.
 
-cmake_minimum_required(VERSION 2.8.8)
+cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
+project(KaaCppSdk)
+enable_language(CXX)
```
Comment (rasendubi) [inline client/client-multi/client-cpp/CMakeLists.txt]: Use `project(KaaCppSdk C CXX)` syntax.

File: client/client-multi/client-cpp/CMakeLists.txt [hunk 2]
```diff
@@ -208,7 +207,7 @@ endif()
 # NOTE: Platform specific compilation flags should be set 
 # in the corresponding platform CMake script(s).
 if (NOT MSVC)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+    list(APPEND KAA_COMPILE_OPTIONS -fPIC)
```
Comment (rasendubi) [inline client/client-multi/client-cpp/CMakeLists.txt]: Have you tested with non-msvc compilers? `list(APPEND` is not suitable for compiler flags as it inserts `;` and that causes cmake interpret `-fPIC` as a name for executable.

File: client/client-multi/client-cpp/CMakeLists.txt [hunk 3]
```diff
@@ -217,12 +216,16 @@ endif ()
 
 #Compiler specific flags
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=gnu++11 -stdlib=libc++")
+    list(APPEND KAA_COMPILE_OPTIONS
```
Comment (rasendubi) [inline client/client-multi/client-cpp/CMakeLists.txt:209]: Same here.

File: client/client-multi/client-cpp/CMakeLists.txt [hunk 4]
```diff
@@ -217,12 +216,16 @@ endif ()
...
+    list(APPEND KAA_COMPILE_OPTIONS 
+                    -Wall
+                    -std=gnu++11
+                    -stdlib=libc++)
 endif()
 
-
 if (CMAKE_COMPILER_IS_GNUCXX)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++0x") 
+    list(APPEND KAA_COMPILE_OPTIONS
```
Comment (rasendubi) [inline client/client-multi/client-cpp/CMakeLists.txt:216]: And here

File: client/client-multi/client-cpp/CMakeLists.txt [hunk 5]
```diff
@@ -256,8 +259,9 @@ message("KAA_MAX_LOG_LEVEL=${KAA_MAX_LOG_LEVEL}")
 
 if ( NOT KAA_WITHOUT_EVENTS )
```
Comment (forGGe) [inline client/client-multi/client-cpp/CMakeLists.txt]: Prefer use consistent cmake style. Here and below different policies are used for whitespaces

File: client/client-multi/client-cpp/CMakeLists.txt (hunk 2 above)
Comment (ghost) [inline client/client-multi/client-cpp/CMakeLists.txt]: It works well both on 
gcc (Ubuntu 4.8.5-2ubuntu1~14.04.1) 4.8.5 
and
MSVC C/C++ Optimizing Compiler Version 19.00.23506 for x86

Comment (ghost) [inline client/client-multi/client-cpp/CMakeLists.txt]: By the way, I am going to remove fPIC because providing only single type library (static|shared)

**Comment by forGGe** [comment]
Looks good
@rasendubi  please review

**Pull request merged**